### PR TITLE
DataTransformer uses proper data.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ x ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ x ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Autowire DataTransformer classes. Provide DataTransformer classes with types.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. See if the build completes.

### CODE REVIEW

**Review by kuzmany — APPROVED** [review]
Do not see issues here 👍 
Good job

File: app/bundles/WebhookBundle/Form/DataTransformer/EventsToArrayTransformer.php [hunk 1]
```diff
@@ -17,7 +20,9 @@ public function __construct(
     /**
      * Convert from the PersistentCollection of Event entities to a simple array.
      *
-     * @return array
+     * @phpstan-param $events Collection<int, Event>
```
Comment (escopecz) [inline app/bundles/WebhookBundle/Form/DataTransformer/EventsToArrayTransformer.php]:
```suggestion
     * @param $events Collection<int, Event>
```
I don't see a reason for phpstan-param.

File: app/bundles/WebhookBundle/Form/DataTransformer/EventsToArrayTransformer.php [hunk 2]
```diff
@@ -32,13 +37,14 @@ public function transform($events)
     /**
      * Convert a simple array into a PersistentCollection of Event entities.
      *
-     * @return PersistentCollection
+     * @phpstan-param $submittedArray array<int, string>
```
Comment (escopecz) [inline app/bundles/WebhookBundle/Form/DataTransformer/EventsToArrayTransformer.php]:
```suggestion
     * @param $submittedArray array<int, string>
```

**Review by escopecz — COMMENTED** [review]
Looks good. 2 questions

File: app/bundles/WebhookBundle/Form/DataTransformer/EventsToArrayTransformer.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/WebhookBundle/Form/DataTransformer/EventsToArrayTransformer.php]: Correct, sorry, i'm used to have those.

File: app/bundles/CoreBundle/Form/DataTransformer/SortableListTransformer.php [hunk 3]
```diff
@@ -41,7 +44,12 @@ public function reverseTransform($array)
         return $this->formatList($array);
     }
 
-    private function formatList($array)
+    /**
+     * @param array<mixed>|null $array
+     *
+     * @return array<mixed>
+     */
+    private function formatList(?array $array): array
```
Comment (escopecz) [inline app/bundles/CoreBundle/Form/DataTransformer/SortableListTransformer.php:52]: I started a discussion about changing method signature and if we should still consider it a BC break:

https://mautic.slack.com/archives/CQMKV0RU1/p1704705746283329

Comment (escopecz) [reply app/bundles/CoreBundle/Form/DataTransformer/SortableListTransformer.php:52]: I missed that this is a private method. Not a BC break

**Review by escopecz — APPROVED** [review]
All good from the code perspective. Thank you! 👍

**Pull request merged**